nemo/core/callbacks.py

```diff
@@ -37,6 +37,60 @@
 logging = nemo.logging
 
 
+class NeMoCallback(ABC):
+    def __init__(self):
+        self._action = None
+
+    @property
+    def action(self):
+        """TODO remove"""
+        return self._action
+
+    @action.setter
+    def action(self, action_obj):
+        self._action = action_obj
+
+    def on_action_start(self, state):
+        pass
+
+    def on_action_end(self, state):
+        pass
+
+    def on_epoch_start(self, state):
+        pass
+
+    def on_epoch_end(self, state):
+        pass
+
+    def on_iteration_start(self, state):
+        pass
+
+    def on_iteration_end(self, state):
+        pass
+
+
+class SimpleLossLogger(NeMoCallback):
+    def __init__(self, step_freq=100, tensors_to_log=["loss"]):
+        super().__init__()
+        # Step_freq: how often logs are printed
+        self.step_freq = step_freq
+        self.tensors_to_log = tensors_to_log
+
+    # def on_optimizer_step_stop(self, state, tensors_to_log=[“loss”]):
+    #     #tensors_to_log: List of keys into state that will be logged
+
+    def on_iteration_end(self, state):
+        if state["step"] % self.step_freq == 0:
+            for tensor_key in self.tensors_to_log:
+                tensor = state["tensors"].get_tensor(tensor_key)
+                if tensor is None:
+                    tensor = state["tensors"].get_and_compute_tensor(tensor_key, self.action)
+                logging.info("%s: %s", tensor_key, tensor)
+                # except KeyError:
+                #     raise KeyError(f"{self} was passed {tensor_key} but the tensor was not found in the state_dict. "
+                #                    f"Current state tensors include {state['tensors'].tensor_list()}")
+
+
 class ActionCallback(ABC):
```

Comment:
Is the plan going forward to only have one "base" callback - NeMoCallback?

Reply:
I think the current callback system will be backwards compatible with the old system. So we can have both NeMoCallback and a deprecated ActionCallback.